internal/common/expiring_cache.go

```diff
@@ -0,0 +1,46 @@
+package common
+
```

Comment:
why not to use https://github.com/jellydator/ttlcache , it has everything we need

Reply:
We already use this one, https://github.com/patrickmn/go-cache
I wanted to add GetOrInsert

Reply:
BTW, https://github.com/jellydator/ttlcache uses generics (which is nice), but was added in go 1.18.  We are in 1.17.